### PR TITLE
[WIP] Enable AAD integration

### PR DESCRIPTION
**Pull request by jim-minter**

### CODE REVIEW

**Comment by jim-minter** [comment]
Untested so far.  Maybe it'll be as simple as this, maybe not.

File: examples/openshift.json [hunk 1]
```diff
@@ -58,10 +56,6 @@
           }
         ]
       }
-    },
-    "servicePrincipalProfile": {
-      "clientId": "YOUR CLIENTID",
```
Comment (kargakis) [inline examples/openshift.json]: This change is breaking azure.conf

Comment (jim-minter) [reply examples/openshift.json]: It shouldn't do - if my reading is right the deploy workflow detects an empty servicePrincipalProfile, creates an application object + new service principal, then enriches the config with that new service principal ID and password.  But I haven't tested it yet ;-)

**Comment by kargakis** [comment]
Please reopen upstream :)

**Pull request closed without merging**